problems/problems.go

```diff
@@ -15,6 +51,69 @@ func (t *DefaultDataProvider) GetSets() ([]byte, error) {
 	return ioutil.ReadFile(fileName)
 }
 
+// FindProblem retrieves problem details for a given user and problem ID. For now it only works with
+// Codeforces.
+func (t *DefaultDataProvider) FindProblem(userID string, problemID uint) (*Problem, error) {
+	problem, err := t.findProblem(problemID)
+	if err != nil {
+		return nil, err
+	}
+
+	// NOTE: Assuming this is a Codeforces problem for now
+	externalID := problem.ExternalID
+	contestID, err := strconv.Atoi(externalID[0 : len(externalID)-1])
+	index := externalID[len(externalID)-1:]
+
```

Comment:
Should we check here if the `atoi` conversion worked so that if there is an error on the conversion, we save the call to the API? 

Also, I think there is `Atou` to convert to `uint` directly.

Reply:
Good catch.

I couldn't find that `Atou` function in https://golang.org/pkg/strconv/.

Reply:
Actually, I don't need to convert this string into int, I will pass it to Codeforces, if it's wrong the request will fail.